bug: TestRunInitErrors/controller_failure fails on main — runInit succeeds with bogus containerd socket

## Repro

On a clean checkout of `origin/main` (verified at `0385750`, `fix: pre-flight containerd reachability in dev-init.sh (#388)`):

```bash
git clone https://github.com/eminwux/kukeon
cd kukeon
go test -run TestRunInitErrors/controller_failure ./cmd/kuke/init/...
```

The test passes `viper.Set(config.KUKEON_ROOT_CONTAINERD_SOCKET.ViperKey, filepath.Join(tmp, "missing.sock"))` and runs `RunInit` against a fresh `t.TempDir()` run-path. It expects `RunInit` to return *any* error (the assertion at `cmd/kuke/init/init_test.go:763` is just `err == nil → t.Fatalf("expected error, got nil")`); a follow-up check at `init_test.go:772` further requires the error not be `ErrLoggerNotFound`.

## Observed

```
=== RUN   TestRunInitErrors/controller_failure
    init_test.go:764: expected error, got nil
--- FAIL: TestRunInitErrors/controller_failure (0.09s)
```

`RunInit` returns `nil` even though the socket path `tmp/missing.sock` does not exist. No CI exposure because the test passed at the time of its merge; this is a regression observable locally (and the upstream `make test` job ran on a hosted runner that may differ environmentally).

I reproduced this on `origin/main` *without* my own change applied (stashed my diff, re-ran, same failure), so this is independent of #297 / #390.

## Root cause

Not yet pinpointed. Candidate shortlist for the next picker, ordered most→least likely:

1. `cmd/kuke/init/init.go:362-365` — `controller.NewControllerExec(...).Bootstrap()` may be returning a no-op success when prerequisite state (containerd socket, kukeond image) is unreachable. If `Bootstrap` silently early-returns instead of dialing containerd, the test will observe a nil error.
2. `cmd/kuke/init/init.go:336-348` — `sysuser.EnsureUserGroup` and `installForwardAdmission` run *before* the controller and may be silently no-op'ing on systems where the kukeon group/admission chain already exists from a prior `make dev-init` (worth checking on hosts that have never run init).
3. `cmd/kuke/init/init.go:325-330` — `instance.VerifyOrWrite` against a fresh `t.TempDir()` should at minimum write an instance marker; if it then short-circuits the rest of the flow somewhere downstream, `Bootstrap` may not be reached.
4. The test setup itself at `cmd/kuke/init/init_test.go:739-752` may need `viper.Set` for additional keys that have crept into `runInit` (e.g. `config.KUKEOND_SOCKET`, `config.KUKEON_ROOT_NAMESPACE_SUFFIX`, `config.KUKEON_ROOT_CGROUP_ROOT`) since the test was written — if a `Load`/`Configure` step ahead of the controller now bails early, the original "controller failure" assertion never gets exercised.

## Suggested fix

Start by reproducing locally and adding a `t.Logf` of the actual `err` (and the report from `Bootstrap` when err is nil) at `init_test.go:763` to see whether `Bootstrap` returned a partial-success report or whether the early `serverconfig.Load`/`ConfigureRuntime`/`instance.VerifyOrWrite` steps short-circuited the controller call entirely. Once the silent-success path is identified, either (a) tighten the production code so a missing `ContainerdSocket` surfaces as an explicit error before bootstrap reports success, or (b) update the test to reach `Bootstrap` deterministically (e.g. by injecting a faked controller via the same MockClientKey pattern the `cmd/kuke/daemon/start` tests already use). Option (a) is the right fix if there is a real production hole; option (b) is acceptable only if no production-observable bug exists and the test is purely defending a guard that has been moved.

--- a/cmd/kuke/init/init.go
+++ b/cmd/kuke/init/init.go
@@ -45,6 +45,12 @@ const (
 	kukeondReadyTimeout = 30 * time.Second
 	kukeondReadyTick    = 200 * time.Millisecond
 
+	// defaultContainerdSocket is the conventional path of a standalone
+	// containerd's gRPC socket. Used as the fallback when no socket is
+	// configured (matches the containerd library's own default and the
+	// pre-flight in scripts/dev-init.sh).
+	defaultContainerdSocket = "/run/containerd/containerd.sock"
+
 	// File modes applied by `kuke init` so the kukeon group can reach the
 	// runtime/socket without world access. Writes under /opt/kukeon still
 	// require root and go through the daemon.
@@ -243,6 +249,29 @@ func resolveKukeondImage() string {
 	return fmt.Sprintf("%s:%s", repo, tag)
 }
 
+// preflightContainerdSocket fails fast when the configured containerd socket
+// does not exist or is not a socket file. Without this check, RunInit can
+// reach the controller bootstrap and silently succeed against an unreachable
+// socket — runner.ExistsRealmContainerdNamespace deliberately returns
+// (false, nil) on a missing socket for test ergonomics, which lets the
+// "namespace missing, create it" path proceed through downstream no-ops.
+func preflightContainerdSocket(path string) error {
+	info, err := os.Stat(path)
+	if err != nil {
+		return fmt.Errorf(
+			"%w: %s — start containerd before re-running `kuke init`: %w",
+			errdefs.ErrConnectContainerd, path, err,
+		)
+	}
+	if info.Mode()&os.ModeSocket == 0 {
+		return fmt.Errorf(
+			"%w: %s is not a socket file — verify containerd is running and the path is correct",
+			errdefs.ErrConnectContainerd, path,
+		)
+	}
+	return nil
+}
+
 // installForwardAdmission installs the kukeon-owned FORWARD admission chain
 // so cells reach the network on hosts where `iptables -P FORWARD DROP` is
 // the default (Docker, firewalld, ufw, hardened distros). Idempotent —
@@ -322,6 +351,14 @@ func runInit(cmd *cobra.Command, _ []string) error {
 		runPath = config.DefaultRunPath()
 	}
 
+	containerdSocket := viper.GetString(config.KUKEON_ROOT_CONTAINERD_SOCKET.ViperKey)
+	if containerdSocket == "" {
+		containerdSocket = defaultContainerdSocket
+	}
+	if err = preflightContainerdSocket(containerdSocket); err != nil {
+		return err
+	}
+
 	if mismatchErr := instance.VerifyOrWrite(
 		runPath,
 		viper.GetString(config.KUKEON_ROOT_NAMESPACE_SUFFIX.ViperKey),
@@ -349,7 +386,7 @@ func runInit(cmd *cobra.Command, _ []string) error {
 
 	opts := controller.Options{
 		RunPath:              runPath,
-		ContainerdSocket:     viper.GetString(config.KUKEON_ROOT_CONTAINERD_SOCKET.ViperKey),
+		ContainerdSocket:     containerdSocket,
 		KukeondImage:         image,
 		KukeondSocket:        socketPath,
 		KukeondSocketGID:     ensure.GID,

--- a/cmd/kuke/init/init_test.go
+++ b/cmd/kuke/init/init_test.go
@@ -736,18 +736,32 @@ func TestRunInitErrors(t *testing.T) {
 			expectError: true,
 		},
 		{
-			name: "controller failure",
+			name: "unreachable containerd socket",
 			setup: func(t *testing.T, cmd *cobra.Command) {
 				t.Cleanup(viper.Reset)
 				tmp := t.TempDir()
 				viper.Set(config.KUKEON_ROOT_RUN_PATH.ViperKey, tmp)
 				viper.Set(config.KUKEON_ROOT_CONTAINERD_SOCKET.ViperKey, filepath.Join(tmp, "missing.sock"))
+				// Point at a non-existent ServerConfiguration so Load
+				// returns an empty doc and applyServerConfiguration does
+				// not override the bogus socket above with whatever lives
+				// in /etc/kukeon/kukeond.yaml on the test host.
+				viper.Set(
+					config.KUKE_INIT_SERVER_CONFIGURATION.ViperKey,
+					filepath.Join(tmp, "missing.yaml"),
+				)
+				// Re-seed package-init defaults that an earlier
+				// t.Cleanup(viper.Reset) in this package will have
+				// wiped — consts.ConfigureRuntime rejects empty values.
+				viper.Set(config.KUKEON_ROOT_NAMESPACE_SUFFIX.ViperKey, config.KUKEON_ROOT_NAMESPACE_SUFFIX.Default)
+				viper.Set(config.KUKEON_ROOT_CGROUP_ROOT.ViperKey, config.KUKEON_ROOT_CGROUP_ROOT.Default)
 				logger := slog.New(slog.NewTextHandler(io.Discard, nil))
 				ctx := context.WithValue(context.Background(), types.CtxLogger, logger)
 				cmd.SetContext(ctx)
 				cmd.SetOut(&bytes.Buffer{})
 				cmd.SetErr(&bytes.Buffer{})
 			},
+			wantErr:     errdefs.ErrConnectContainerd,
 			expectError: true,
 		},
 	}
@@ -768,9 +782,6 @@ func TestRunInitErrors(t *testing.T) {
 			}
 			if tc.wantErr != nil && !errors.Is(err, tc.wantErr) {
 				t.Fatalf("expected error %v, got %v", tc.wantErr, err)
-			}
-			if tc.name == "controller failure" && (err == nil || errors.Is(err, errdefs.ErrLoggerNotFound)) {
-				t.Fatalf("controller failure case should propagate controller error, got %v", err)
 			}
 		})
 	}

